sdk/python/tests/integration/feature_repos/universal/data_sources/bigquery.py

```diff
@@ -12,21 +13,27 @@
 
 
 class BigQueryDataSourceCreator(DataSourceCreator):
+    dataset: Optional[Dataset] = None
+
     def __init__(self, project_name: str):
         self.client = bigquery.Client()
         self.project_name = project_name
         self.gcp_project = self.client.project
         self.dataset_id = f"{self.gcp_project}.{project_name}"
-        self.dataset = bigquery.Dataset(self.dataset_id)
-        print(f"Creating dataset: {self.dataset_id}")
-        self.client.create_dataset(self.dataset, exists_ok=True)
-        self.dataset.default_table_expiration_ms = (
-            1000 * 60 * 60 * 24 * 14
-        )  # 2 weeks in milliseconds
-        self.client.update_dataset(self.dataset, ["default_table_expiration_ms"])
 
         self.tables = []
 
+    def get_dataset(self):
+        if not self.dataset:
+            self.dataset = bigquery.Dataset(self.dataset_id)
+            print(f"Creating dataset: {self.dataset_id}")
+            self.client.create_dataset(self.dataset, exists_ok=True)
+            self.dataset.default_table_expiration_ms = (
+                1000 * 60 * 60 * 24 * 14
+            )  # 2 weeks in milliseconds
+            self.client.update_dataset(self.dataset, ["default_table_expiration_ms"])
+        return self.dataset
```

Comment:
nit: create_dataset instead? and doesn't seem like you use the returned self.dataset anywhere

Reply:
Yeah I can change